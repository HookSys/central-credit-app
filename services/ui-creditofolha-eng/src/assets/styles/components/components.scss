@import "input";
@import "progress";
@import "tooltip";
@import "spinner";
@import "avatar";
@import "dropdown";
@import "cards";
@import "grid";
@import "toast";
