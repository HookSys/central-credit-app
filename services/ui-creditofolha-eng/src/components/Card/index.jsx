import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import GridLayout from 'components/GridLayout'

const Card = ({ children, className, size }) => {
  const cardTemplate = `
    'title actions'
    'infos infos'
    / 1fr 1fr
  `

  return (
    <div className={ size }>
      <GridLayout
        gridTemplate={ cardTemplate }
        className={ classNames('card', className) }
      >
        { children }
      </GridLayout>
    </div>
  )
}

Card.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  size: PropTypes.string,
}

Card.defaultProps = {
  className: '',
  size: 'col-12 col-sm-6 col-xl-4',
}

export default Card
