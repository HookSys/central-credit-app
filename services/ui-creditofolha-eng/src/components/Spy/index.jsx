import { useHistory } from 'react-router-dom'
import { useSelector, useDispatch } from 'react-redux'

// variable which will point to react-router history
let globalHistory = null
let globalDispatch = null
let globalAccess = {}

const Spy = () => {
  const access = useSelector((state) => state.auth.get('access'))
  const userFunction = useSelector((state) => {
    const entity = state.user.get('data').getSelectedEntity()
    return !entity ? null : entity.get('identificador')
  })
  const dispatch = useDispatch()
  const history = useHistory()

  globalHistory = history
  globalDispatch = dispatch
  globalAccess = {
    access,
    userFunction,
  }

  return null
}

export default Spy

// export react-router history
export function getHistory() {
  return globalHistory
}

export function getDispatch() {
  return globalDispatch
}

export function getAccess() {
  return globalAccess
}
