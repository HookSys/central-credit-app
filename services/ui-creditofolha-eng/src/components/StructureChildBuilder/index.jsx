import React from 'react'
import { Route, Switch } from 'react-router-dom'
import MetaTags from 'components/MetaTags'
import Permissions from 'components/Permissions'
import DefaultContainer from 'components/DefaultContainer'

const StructureChildBuilder = (structure, route, rootPath) => {
  const { ROUTES, NAME: ChildName, COMPONENT } = route
  const Container = COMPONENT || DefaultContainer
  return Object.keys(ROUTES).reverse().map((key) => {
    const path = key === 'INDEX' ? '' : `${ rootPath }${ ROUTES[key].URL }`
    const Page = ROUTES[key].COMPONENT
    const permissions = ROUTES[key].VALIDATION
    const isFeedback = ROUTES[key].IS_FEEDBACK
    const hasChilds = typeof ROUTES[key].ROUTES === 'object'

    if (isFeedback) {
      return (
        <Route exact path={ path } key={ ChildName + key }>
          <Permissions permissions={ permissions }>
            <MetaTags
              metaTitle={ ROUTES[key].NAME }
              metaTitleSuffix={ ChildName }
            />
            <Page structure={ ROUTES[key] } rootPath={ rootPath } parentStructure={ route } />
          </Permissions>
        </Route>
      )
    }

    if (hasChilds) {
      return (
        <Route path={ path } key={ ChildName + key }>
          <Permissions permissions={ permissions }>
            <Switch>
              { StructureChildBuilder(structure, ROUTES[key], path) }
            </Switch>
          </Permissions>
        </Route>
      )
    }

    return (
      <Route exact path={ path } key={ ChildName + key }>
        <Permissions permissions={ permissions }>
          <Container structure={ route } rootPath={ rootPath } parentStructure={ structure }>
            <MetaTags
              metaTitle={ ROUTES[key].NAME }
              metaTitleSuffix={ ChildName }
            />
            <Page structure={ ROUTES[key] } rootPath={ rootPath } parentStructure={ route } />
          </Container>
        </Permissions>
      </Route>
    )
  })
}

export default StructureChildBuilder
