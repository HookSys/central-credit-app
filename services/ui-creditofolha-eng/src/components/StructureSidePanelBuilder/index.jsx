
import React from 'react'
import { Route } from 'react-router-dom'
import MetaTags from 'components/MetaTags'
import Permissions from 'components/Permissions'

const StructureSidePanelBuilder = (structure, route, rootPath) => {
  const { SIDEPANEL_ROUTES, NAME: ChildName, COMPONENT: Container } = route
  return Object.keys(SIDEPANEL_ROUTES).reverse().map((key) => {
    const path = key === 'INDEX' ? '' : `${ rootPath }${ SIDEPANEL_ROUTES[key].URL }`
    const Page = SIDEPANEL_ROUTES[key].COMPONENT
    const permissions = SIDEPANEL_ROUTES[key].VALIDATION
    const isFeedback = SIDEPANEL_ROUTES[key].IS_FEEDBACK

    if (isFeedback) {
      return (
        <Route exact path={ path } key={ ChildName + key }>
          <Permissions permissions={ permissions }>
            <MetaTags
              metaTitle={ SIDEPANEL_ROUTES[key].NAME }
              metaTitleSuffix={ ChildName }
            />
            <Page
              structure={ SIDEPANEL_ROUTES[key] }
              rootPath={ rootPath }
              parentStructure={ route }
            />
          </Permissions>
        </Route>
      )
    }

    return (
      <Route exact path={ path } key={ ChildName + key }>
        <Permissions permissions={ permissions }>
          <Container
            structure={ route }
            rootPath={ rootPath }
            parentStructure={ structure }
          >
            <MetaTags
              metaTitle={ SIDEPANEL_ROUTES[key].NAME }
              metaTitleSuffix={ ChildName }
            />
            <Page
              structure={ SIDEPANEL_ROUTES[key] }
              rootPath={ rootPath }
              parentStructure={ route }
            />
          </Container>
        </Permissions>
      </Route>
    )
  })
}

export default StructureSidePanelBuilder
