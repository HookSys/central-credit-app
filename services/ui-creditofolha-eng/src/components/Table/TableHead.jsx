import React from 'react'
import PropTypes from 'prop-types'

export const TableHead = ((props) => {
  const { children, className } = props
  return (
    <thead>
      <tr className={ className }>
        { children }
      </tr>
    </thead>
  )
})

TableHead.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
}

TableHead.defaultProps = {
  className: '',
}
