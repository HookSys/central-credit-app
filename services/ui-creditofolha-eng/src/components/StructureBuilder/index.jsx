import React from 'react'
import { Route, Switch } from 'react-router-dom'
import { StructureRoutesBuilder, Permissions, ThemeRender } from 'components'

const StructureBuilder = (structure) => {
  const { CONTAINER: Container, VALIDATION, THEME } = structure

  return (
    <Route path={ structure.ENTRY }>
      <Permissions permissions={ VALIDATION }>
        <ThemeRender theme={ THEME }>
          <Container structure={ structure }>
            <Switch>
              { StructureRoutesBuilder(structure) }
            </Switch>
          </Container>
        </ThemeRender>
      </Permissions>
    </Route>
  )
}

export default StructureBuilder
