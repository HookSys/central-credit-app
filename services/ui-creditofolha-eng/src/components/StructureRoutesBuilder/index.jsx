import React from 'react'
import { Route, Switch } from 'react-router-dom'
import { STRUCTURE_TYPES } from 'constants/structure'
import MetaTags from 'components/MetaTags'
import Permissions from 'components/Permissions'
import StructureChildBuilder from '../StructureChildBuilder'
import StructureSidePanelBuilder from '../StructureSidePanelBuilder'

const StructureRoutesBuilder = (structure) => {
  const { type, entry, routes, name: ContainerName } = structure
  const rootPath = type !== STRUCTURE_TYPES.DEFAULT ? entry : ''
  return Object.keys(routes).reverse().map((key) => {
    const path = key === 'INDEX' ? '' : `${ rootPath }${ routes[key].route }`
    const permissions = null
    const hasChilds = typeof routes[key].routes === 'object'
    const hasSidePanelRoutes = typeof routes[key].SIDEPANEL_ROUTES === 'object'
    const Page = routes[key].component

    if (hasChilds) {
      return (
        <Route path={ path } key={ entry + key }>
          <Permissions permissions={ permissions }>
            <Switch>
              { StructureChildBuilder(structure, routes[key], path) }
            </Switch>
          </Permissions>
        </Route>
      )
    }

    if (hasSidePanelRoutes) {
      return (
        <Route path={ path } key={ entry + key }>
          <Permissions permissions={ permissions }>
            <Switch>
              { StructureSidePanelBuilder(structure, routes[key], path) }
            </Switch>
          </Permissions>
        </Route>
      )
    }

    return (
      <Route exact path={ path } key={ entry + key }>
        <Permissions permissions={ permissions }>
          <MetaTags
            metaTitle={ routes[key].name }
            metaTitleSuffix={ ContainerName }
          />
          <Page structure={ routes[key] } rootPath={ rootPath } parentStructure={ structure } />
        </Permissions>
      </Route>
    )
  })
}

export default StructureRoutesBuilder
