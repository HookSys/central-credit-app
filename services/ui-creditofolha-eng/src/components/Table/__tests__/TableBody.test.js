
import React from 'react'
import { render } from '@testing-library/react'
import { TableBody } from 'components/Table'

test('should render table body component', () => {
  const { container } = render(
    <table>
      <TableBody>
        <tr>
          <td>
            TableBody test
          </td>
        </tr>
      </TableBody>
    </table>
  )
  const tree = container.firstChild
  expect(tree).toMatchSnapshot()
})
