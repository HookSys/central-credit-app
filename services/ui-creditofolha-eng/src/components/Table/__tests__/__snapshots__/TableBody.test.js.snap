// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render table body component 1`] = `
<table>
  <tbody
    class=""
  >
    <tr>
      <td>
        TableBody test
      </td>
    </tr>
  </tbody>
</table>
`;
